src/main/java/de/tum/in/www1/artemis/web/rest/TextAssessmentResource.java

```diff
@@ -137,15 +144,32 @@ public ResponseEntity cancelAssessment(@PathVariable Long exerciseId, @PathVaria
         return ResponseEntity.ok().build();
     }
 
-    @Transactional
+    @Transactional(readOnly = true)
     @GetMapping("/result/{resultId}/with-textblocks")
     @PreAuthorize("hasAnyRole('TA', 'INSTRUCTOR', 'ADMIN')")
     public ResponseEntity<Result> getResultWithPredefinedTextblocks(@PathVariable Long resultId) throws EntityNotFoundException, AccessForbiddenException {
         final Result result = resultService.findOneWithSubmission(resultId);
         final Exercise exercise = result.getParticipation().getExercise();
         checkAuthorization(exercise);
 
-        textBlockService.prepopulateFeedbackBlocks(result);
+        if (!(exercise instanceof TextExercise)) {
+            throw new BadRequestAlertException("No text exercise found for the given ID.", "textExercise", "exerciseNotFound");
+        }
+
+        final TextExercise textExercise = (TextExercise) exercise;
+
+        if (automaticTextFeedbackService.isPresent() && textExercise.isAutomaticAssessmentEnabled()) {
+            automaticTextFeedbackService.get().suggestFeedback(result);
+        }
+        else {
+            textBlockService.prepopulateFeedbackBlocks(result);
+        }
+
+        Comparator<TextBlock> byStartIndexReversed = (TextBlock first, TextBlock second) -> Integer.compare(second.getStartIndex(), first.getStartIndex());
+        TextSubmission textSubmission = (TextSubmission) result.getSubmission();
+        textSubmission.getBlocks().sort(byStartIndexReversed);
+
+        result.getParticipation().setStudent(null);
```

Comment:
could be problematic with the `@Transactional` but I'm not really sure about this

Reply:
I removed the `@Transactional` by pulling in @SiggZ commit 7742307 from #656.